app-nodejs-html5/src/main/webapp/app.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var keycloak = new Keycloak();
+var serviceUrl = 'http://127.0.0.1:3000/service'
+
+function notAuthenticated() {
+    document.getElementById('not-authenticated').style.display = 'block';
+    document.getElementById('authenticated').style.display = 'none';
+}
+
+function authenticated() {
+    document.getElementById('not-authenticated').style.display = 'none';
+    document.getElementById('authenticated').style.display = 'block';
+    document.getElementById('message').innerHTML = 'User: ' + keycloak.tokenParsed['preferred_username'];
+}
```

Comment:
Since the queried elements here are static in the DOM you can store them in a variable so they only have to be queried once.
```suggestion
const notAuthenticatedElement = document.getElementById('not-authenticated');
const authenticatedElement = document.getElementById('authenticated');
const messageElement = document.getElementById('message');

function notAuthenticated() {
    notAuthenticatedElement.style.display = 'block';
    authenticatedElement.style.display = 'none';
}

function authenticated() {
    notAuthenticatedElement.style.display = 'none';
    authenticatedElement.style.display = 'block';
    messageElement.innerHTML = 'User: ' + keycloak.tokenParsed['preferred_username'];
}
```